packages/waku/src/router/create-pages.ts

```diff
@@ -312,7 +326,15 @@ export const createPages = <
     path: string,
     method: string,
   ) => string | undefined = (path, method) => {
-    for (const [p, v] of apiPathMap.entries()) {
+    const apiConfigEntries = Array.from(apiPathMap.entries()).sort(
+      ([__, a], [___, b]) => {
+        return (
+          getRoutePriority({ path: a.pathSpec, type: 'api' }) -
+          getRoutePriority({ path: b.pathSpec, type: 'api' })
+        );
+      },
+    );
+    for (const [p, v] of apiConfigEntries) {
```

Comment:
The main bug here is that the `apiPathMap` is sorted by insertion order rather than priority order - a potentially more efficient solution here is to use a different data structure that preserves priority order on insertion (e.g. some kind of tree or trie)

Reply:
> The main bug here is that the apiPathMap is sorted by insertion order rather than priority order

what exactly do you mean by this? The `Map` isn't ordered.

Reply:
Yea at the moment i'm leaning towards the first option (keep maps as-is, sort when doing path matching) to be the least invasive change, haven't had much time to get back to this but hopefully this weekend i will have some time.